### PR TITLE
Azure Monitor Exporter - Set RoleName/RoleInstance from Activity.Resource

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Parsed `activity.GetResource()` to generate role name and instance.

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs [hunk 1]
```diff
@@ -82,22 +84,59 @@ internal async ValueTask<int> AddBatchActivityAsync(Batch<Activity> batchActivit
...
             }
-
-            // TODO: "ai.location.ip"
             // TODO: Handle exception
             telemetryItem.Tags[ContextTagKeys.AiInternalSdkVersion.ToString()] = SdkVersionUtils.SdkVersion;
 
             return telemetryItem;
         }
 
+        internal static void GetRoleInfo(Resource resource, out string roleName, out string roleInstance)
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:101]: minor - in general, if a method is returning void, then it shouldn't be named "Get..". I think ExtractRoleInfo, or PopulatRoleInfoFromResource are better.

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
No an issue introduced in this exporter, but storing the Resource in every Activity, extracting it (which involves string allocations) is not good. Need to design something better for this in OTel SDK itself.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs [hunk 2]
```diff
@@ -82,22 +84,59 @@ internal async ValueTask<int> AddBatchActivityAsync(Batch<Activity> batchActivit
...
+                    serviceName = attribute.Value.ToString();
+                }
+                else if (attribute.Key == Resource.ServiceNamespaceKey && attribute.Value is string)
+                {
+                    serviceNamespace = attribute.Value.ToString();
+                }
+                else if (attribute.Key == Resource.ServiceInstanceIdKey && attribute.Value is string)
+                {
+                    roleInstance = attribute.Value.ToString();
+                }
```
Comment (TimothyMothra) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs:127]: i see that you're doing the `attribute.Value is string` three times.
consider putting that in an outer if statement and the key compare in an inner if.

**Pull request merged**